timescaledb/tutorials/analyze-intraday-stocks/fetch-and-ingest.md

```diff
@@ -0,0 +1,230 @@
+# 2. Fetch and insert intraday stock data
+
+In this step, you will
+
+* create a configuration file (optional)
+* fetch stock data
+* insert the data into TimescaleDB
+
+## Create a configuration file
+
+This is an optional step, but it is highly recommended not to store your password or other sensitive information
+directly in your code. Instead, create a configuration file, for example `config.py`, and include your 
+database connection details and Alpha Vantage API key in there:
+
+```python
+# example content of config.py:
+DB_USER = 'user'
+DB_PASS = 'passwd'
+DB_HOST = 'host'
+DB_PORT = '000'
+DB_NAME = 'db'
+APIKEY = 'alpha_vantage_apikey'
+```
+
+Later, whenever you need to reference any of the information from this config file, you just need to import it:
+```python
+import config
+apikey = config.APIKEY
+...
+```
+
+## Collect ticker symbols
+
+In order to fetch intraday stock data, you will need to know which ticker symbols you want to analyze.
+
+First, let's collect a list of symbols so later we can fetch their data!
+
+In general, you have a few options to gather a list of ticker symbols dynamically:
+
+* Scrape it from a public website (example code here).
+* Use an API that has this functionality.
+* Download it from an open repository
+
+To keep things simple, just download this CSV file to get started:
+
+* [Download top 100 US ticker symbols (based on market capitalization)]()
+
+## Read symbols from CSV file
+
+After downloading the CSV file into the project folder, read the symbols into a list:
+
+```python
+import csv
+with open('ticker_symbols.csv') as f:
+    reader = csv.reader(f)
+    symbols = [row[0] for row in reader]
+    print(symbols)
+
+['MMM', 'AOS', 'ABT', 'ABBV', 'ABMD']
+```
+
+Now you have a list of ticker symbols that you can use later to make requests to the Alpha Vantage API.
+
+## Fetching intraday stock data
+
+*It's not necessary to know how Alpha Vantage API works to complete this tutorial, so feel free to skip this section below, if you want.*
+
+### About the API
+
+Alpha Vantage API provides 2-year historical intraday stock data in 1-min, 5-min, 15-min, or 30-min 
+intervals.
+
+Because the API outputs a lot of data in a CSV file (~2200 rows/symbol/day in case of 1-min interval), it
+slices up the dataset into one-month buckets. This means, for one request the most amount of data you can get for one symbol is one month.
+
+To fetch the maximum amount of historical intraday data (last 24 months), you need to slice up your requests by month (year1month1, year1month2, etc...).
+
+Also, keep in mind that one request can only fetch data for one symbol.
```

Comment:
```suggestion
Alpha Vantage API provides 2 year historical intraday stock data in 1, 5, 15, or 30 minute 
intervals. The API outputs a lot of data in a CSV file (around 2200 rows per symbol per 
day, for a 1 minute interval), so it slices the dataset into one month buckets. This means 
that for one request for a single symbol, the most amount of data you can get is one month.
 The maximum amount of historical intraday data is 24 months. To fetch the maximum 
 amount, you need to slice up your requests by month. For example, `year1month1`, 
 `year1month2`, and so on. Keep in mind that each request can only fetch data for one 
 symbol at a time.
```

Don't be afraid to have more than one sentence in a paragraph! It helps readers read if you throw in a longer one every so often, especially if you've just had a long section of single sentences and code snippets.